docs/tutorials/gluon/logistic_regression_explained.md

```diff
@@ -0,0 +1,215 @@
+
+
+```python
+loss = gluon.loss.SigmoidBinaryCrossEntropyLoss()
+trainer = Trainer(params=net.collect_params(), optimizer='sgd', 
+                  optimizer_params={'learning_rate': 0.1, "wd": 0.01})
+accuracy = mx.metric.Accuracy()
+```
+
+Usually, it is not enough to pass the training data through a network only once to achieve high Accuracy. It helps when the network sees each example multiple times. The number of displaying every example to the network is called `epoch`. How big this number should be is unknown in advance, and usually it is estimated using trial and error approach. 
+
+Below we are defining the main training loop, which go through each example in batches specified number of times (epochs). After each epoch we display training loss, validation loss and calculate accuracy of the model using validation set. For now, let's take a look into the code, and I am going to explain the details later.
+
+
+```python
+epochs = 10
+threshold = 0.5
+
+# iterate over training data epochs number of times
+for e in range(epochs):
+    cumulative_train_loss = 0
+    cumulative_val_loss = 0
+    
+    # iterate over all batches of training data and calculate training loss
+    for i, (data, label) in enumerate(train_dataloader):
+        with autograd.record():
+            # Do forward pass on a batch of training data
+            output = net(data)
+        
+            # Calculate loss for the training data batch
+            loss_result = loss(output, label)
+
+        # Calculate gradients 
+        loss_result.backward()
+
+        # Change parameters of the network
+        trainer.step(batch_size)
+
+        # Since we calculate loss per single batch, but want to display it per epoch
+        # we sum losses of every batch per an epoch into a single variable
+        cumulative_train_loss += nd.sum(loss_result).asscalar()
+
+    # iterate over all batches of validation data and calculate validation loss
+    for i, (val_data, val_ground_truth_class) in enumerate(val_dataloader):
+        # Do forward pass on a batch of validation data
+        output = net(val_data)
+        
+        # Similar to cumulative training loss, calculate cumulative validation loss
+        cumulative_val_loss += nd.sum(loss(output, val_ground_truth_class)).asscalar()
+        
+        # Applying sigmoid function, to get data in range [0, 1] and then
+        # subtracting threshold, to make 0 serve as a class boundary: below 0 - class 0, above 0 - class 1
+        # Apply mx.nd.ceil to get classes: convert negative values to 0 and positive to 1.
+        prediction = mx.nd.ceil(net(val_data).sigmoid() - threshold)
+        
+        # Reshape predictions to match dimension of val_ground_truth_class
+        # and update accuracy with the results for that batch.
+        accuracy.update(val_ground_truth_class, prediction.reshape(-1)) 
+    
+    # in the end of epoch, we print out current values for epoch, training and validation losses, and accuracy
```

Comment:
Stating the obvious.

Reply:
Removed the print comment, simplified the `accuracy.update` comment